### PR TITLE
Adding github release request mapper for ripgrep

### PR DESCRIPTION
**Pull request by creedasaurus**
Kind of a start. Not sure if all the same os/arch mappings are there so I default a missing on to linux, atm.

### CODE REVIEW

File: ripgrep/releases.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  s390x: /\b(s390x)/i
+};
+
+const fileExtMap = {
+  deb: /\.deb$/i,
+  pkg: /\.pkg$/i,
+  exe: /\.exe$/i,
+  msi: /\.msi$/i,
+  zip: /\.zip$/i,
+  tar: /\.(tar(\.?(gz)?)|tgz)/i,
```
Comment (solderjs) [inline ripgrep/releases.js:30]: This is too strict for the general case.
There's bz2, gz, xz, and probably others.

**Comment by solderjs** [comment]
Looks good. I'll test it out. Thanks a tonne!

**Comment by solderjs** [comment]
BTW, what does `?!` do in a RegExp?

**Comment by creedasaurus** [comment]
> BTW, what does `?!` do in a RegExp?

Oh yeah, it's a reverse lookup since `x86-64` matches x86 if it happens to catch that first. The reverse lookup checks behind those characters and just makes sure that `x86` is _not_ followed by `-64` or `_64`.

**Pull request merged**